tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/Properties/DefaultCompoundPropertyTest.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+/**
+ * This file is part of contao-community-alliance/dc-general.
+ *
+ * (c) 2013-2017 Contao Community Alliance.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * This project is provided in good faith and hope to be usable by anyone.
+ *
+ * @package    contao-community-alliance/dc-general
+ * @author     Sven Baumann <baumann.sv@gmail.com>
+ * @copyright  2013-2017 Contao Community Alliance.
+ * @license    https://github.com/contao-community-alliance/dc-general/blob/master/LICENSE LGPL-3.0
+ * @filesource
+ */
+
+namespace ContaoCommunityAlliance\DcGeneral\Test\DataDefinition\Definition\Properties;
+
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\Collection\DefaultSubProperty;
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\DefaultCompoundProperty;
+
+/**
+ * This tests DefaultCompoundProperty
+ *
+ * @coversDefaultClass \ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\DefaultCompoundProperty
+ */
+class DefaultCompoundPropertyTest extends DefaultPropertyTest
+{
+    /**
+     * Get a property.
+     *
+     * @param string $name The collection property name.
+     *
+     * @return DefaultCompoundProperty
+     */
+    private function getProperty($name)
+    {
+        return new DefaultCompoundProperty($name);
+    }
+
+    /**
+     * Get a collection property.
+     *
+     * @param string $name The collection property name.
+     *
+     * @return DefaultSubProperty
+     */
+    private function getCollectionProperty($name)
+    {
+        return new DefaultSubProperty($name);
+    }
+
+    /**
+     * Test for clear the properties form the collection.
+     *
+     * @covers ::clearProperties
+     * @covers ::addProperty
+     * @covers ::getProperties
+     * @covers ::hasProperty
+     */
+    public function testClearPropertiesCollection()
+    {
+        $property = $this->getProperty('dummy');
+
+        $this->assertInternalType('array', $property->getProperties());
+
+        $collectionProperty = $this->getCollectionProperty('dummy-collection-property');
+
+        $property->addProperty($collectionProperty);
+
+        $this->assertInternalType('array', $property->getProperties());
```

Comment:
All the assertions here can be simplified as noted above.

Reply:
change